configure.ac

```diff
@@ -4015,12 +4015,6 @@ if test "x$DISABLE_MCS_DOCS" = "xdefault"; then
    DISABLE_MCS_DOCS=$DISABLE_MCS_DOCS_default
 fi
 
-AC_ARG_WITH(lazy_gc_thread_creation, [  --with-lazy-gc-thread-creation=yes|no      Enable lazy runtime thread creation, embedding host must do it explicitly (defaults to no)],[
```

Comment:
How did you determine it was dead?
AFAICT, we still ship it: https://github.com/xamarin/xamarin-macios/blob/0a47a8ce0c6d00e7d36e3315289c99de01f417b2/builds/Makefile#L1263